src/components/HOCs/WithCurrentTask/WithCurrentTask.jsx

```diff
@@ -184,77 +184,79 @@ export const mapDispatchToProps = (dispatch, ownProps) => {
-              ),
-            );
-          } else {
-            dispatch(addTaskComment(taskId, comment, taskStatus));
+      const doAfter = () =>
+        new Promise(async (resolve) => {
+          if (_isString(comment) && comment.length > 0) {
+            if (taskBundle) {
+              await dispatch(
+                addTaskBundleComment(
+                  taskBundle.bundleId,
+                  AsMappableBundle(taskBundle).primaryTaskId() || taskId,
+                  comment,
+                  taskStatus,
+                ),
+              );
+            } else {
+              await dispatch(addTaskComment(taskId, comment, taskStatus));
+            }
           }
-        }
 
-        // Update the user in the background to get their latest score
-        setTimeout(() => dispatch(fetchUser(userId)), 100);
+          // Update the user in the background to get their latest score
+          await dispatch(fetchUser(userId));
 
-        // Updating the challenge actions will allow us to show more accurate
-        // completion progress, but this can be done in the background
-        setTimeout(() => dispatch(fetchChallengeActions(challengeId)), 500);
+          // Updating the challenge actions will allow us to show more accurate
+          // completion progress
+          await dispatch(fetchChallengeActions(challengeId));
 
-        // If working on a virtual challenge, renew it (extend its expiration)
-        // since we've seen some activity, but this can be done in the
-        // background
-        if (_isFinite(ownProps.virtualChallengeId)) {
-          setTimeout(() => dispatch(renewVirtualChallenge(ownProps.virtualChallengeId)), 1000);
-        }
+          // If working on a virtual challenge, renew it (extend its expiration)
+          if (_isFinite(ownProps.virtualChallengeId)) {
+            await dispatch(renewVirtualChallenge(ownProps.virtualChallengeId));
+          }
 
-        if (taskLoadBy) {
-          // Start loading the next task from the challenge.
-          const loadNextTask = _isFinite(requestedNextTask)
-            ? nextRequestedTask(dispatch, ownProps, requestedNextTask)
-            : nextRandomTask(dispatch, ownProps, taskId, taskLoadBy);
+          if (taskLoadBy) {
+            // Start loading the next task from the challenge.
+            const loadNextTask = _isFinite(requestedNextTask)
+              ? await nextRequestedTask(dispatch, ownProps, requestedNextTask)
+              : await nextRandomTask(dispatch, ownProps, taskId, taskLoadBy);
 
-          return loadNextTask
-            .then((newTask) => visitNewTask(dispatch, ownProps, taskId, newTask))
-            .catch(() => {
+            try {
+              await visitNewTask(dispatch, ownProps, taskId, loadNextTask);
+            } catch (error) {
               ownProps.history.push(`/browse/challenges/${challengeId}`);
-            });
-        }
-      };
+            }
+          }
+          resolve();
+        });
```

Comment:
There's a crucial difference between the old `setTimeout` version of this code and the new `async/await` version which I don't think is desired: in the old code, all of the API requests happen in parallel, while in the new code they are constrained to run sequentially (each one can only start once the previous one is finished).

Assuming these tasks are safe to run in parallel, I think you should put the Promises returned by each into an array, and then `await Promise.all(tasks)` at the end, rather than awaiting each one by one.